knowledge-base/grid-filterrow-bool-customization.md

```diff
@@ -0,0 +1,150 @@
+                    </DropDownListSettings>
+                </TelerikDropDownList>
+
+                <!-- Button to clear the filter -->
+                <TelerikButton ButtonType="ButtonType.Button"
+                               Class="k-clear-button-visible ml-2"
+                               Icon="@SvgIcon.FilterClear"
+                               Enabled="@(DropDownListValue != "All")"
+                               OnClick="@(async () =>
+                                          {
+                                              await context.ClearFilterAsync();
+                                              DropDownListValue = "All";
+                                          })">
+                </TelerikButton>
+            </FilterCellTemplate>
+        </GridColumn>
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private List<Employee> GridData { get; set; } = new();
+
+    // Custom options for the dropdown filter
+    private List<string> FilterOptions { get; set; } = new() { "All", "Yes", "No", "No Info" };
+
+    // Selected value in the dropdown filter
+    private string DropDownListValue { get; set; } = "All";
+
+    // Method to apply the filter based on the selected value in the dropdown
+    private async Task ApplyFilter(string newValue, FilterCellTemplateContext context)
+    {
+        DropDownListValue = newValue;
+
+        var filter = context.FilterDescriptor.FilterDescriptors[0] as FilterDescriptor;
+        if (filter != null)
+        {
+            switch (DropDownListValue)
+            {
+                case "Yes":
+                    filter.Value = true;
+                    filter.Operator = FilterOperator.IsEqualTo;
+                    break;
+                case "No":
+                    filter.Value = false;
+                    filter.Operator = FilterOperator.IsEqualTo;
+                    break;
+                case "No Info":
+                    filter.Value = null;
+                    filter.Operator = FilterOperator.IsNull;
+                    break;
+                case "All":
+                    await context.ClearFilterAsync();
+                    return;
+            }
+
+            // Apply the filter
+            await context.FilterAsync();
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        for (int i = 0; i < 30; i++)
+        {
+            GridData.Add(new Employee
+                {
+                    EmployeeId = i,
+                    Name = $"Employee {i}",
+                    IsOnLeave = i % 3 == 0
+                });
+        }
+        GridData.Add(new Employee
+            {
+                EmployeeId = 31,
+                Name = "Employee 31",
+                IsOnLeave = null
+            });
+    }
+
+    public class Employee
+    {
+        public int EmployeeId { get; set; }
+        public string Name { get; set; } = string.Empty!;
+        public bool? IsOnLeave { get; set; }
```

Comment:
This is not something that you must change, but just saying - `IsOnLeave` is not a good candidate for a nullable value. It's highly unlikely that a company doesn't know if an employee is on leave or not. A more suitable nullable boolean parameter can be something like `AgreesToTermsAndConditions` - `true` and `false` or people who already declared it, and `null` for those who haven't said yet.